llvm/lib/Transforms/Scalar/NewGVN.cpp

```diff
@@ -2765,6 +2771,7 @@ NewGVN::makePossiblePHIOfOps(Instruction *I,
       // Clone the instruction, create an expression from it that is
       // translated back into the predecessor, and see if we have a leader.
       Instruction *ValueOp = I->clone();
+      ValueOp->insertBefore(I);
```

Comment:
Can we insert this into PredBB instead and thus avoid the need for the special case?

Reply:
Good point! It makes the code better :)